test/unit/lodash.spec.js

```diff
@@ -0,0 +1,583 @@
+                var str = params[1];
+                expect(str).to.have.length.above(0);
+            });
+            it('should set height value of rect1 to 8', function() {
+                expect(_.set.apply(_, params)).to.eql({
+                    'rectangle': [{
+                        'rect1': {
+                            'width': 4,
+                            'height': 8
+                        }
+                    }]
+                });
+            });
+        });
+
+        describe('set2', function() {
+            beforeEach(function() {
+                params = datasets.set2();
+            });
+            it('should match types of passing elements', function() {
+                var element1 = params[0];
+                var element2 = params[1] instanceof Object;
+                expect(element1).to.be.an('object');
+                expect(element2).to.eql(true);
+                expect(params.length).to.eql(3);
+            });
+            it('pattern should not contain empty string', function() {
+                var str = params[1];
+                expect(str).to.have.length.above(0);
+            });
+            it('should set height value of rect1 to 8', function() {
+                expect(_.set.apply(_, params)).to.eql({
+                    'rectangle': [{
+                        'rect1': {
+                            'width': 4,
+                            'height': 8
+                        }
+                    }]
+                });
+            });
+        });
+    });
+
+    describe('setWith', function() {
+        beforeEach(function() {
+            params = datasets.setWith();
+        });
+        it('should match types of passing elements', function() {
+            var element1 = params[0];
+            var element2 = params[1];
+            var element4 = params[3];
+            expect(element1).to.be.an('object');
+            expect(element2).to.be.a('string');
+            expect(element4).to.be.a('function');
+            expect(params.length).to.eql(4);
+        });
+        it('should set height value of rect1 to 8', function() {
+            // expect(_.set.apply(_, params)).to.eql({ 'rectangle': [{ 'rect1': { 'width': 4, 'height': 8 } }] });
+        });
+    });
+
+    describe('toPairs', function() {
+        beforeEach(function() {
+            params = datasets.toPairs();
+        });
+        it('should be an object', function() {
+            var e1 = params[0];
+            expect(e1).to.be.an('object');
+        });
+        it('should contain property but not own property', function() {
+            var obj = params[0];
+            for (var key in obj) {
+                if (obj.hasOwnProperty(key)) {
+                    expect(key).to.not.eql('red');
+                }
+            }
+            expect(obj).to.have.property('cherry');
+        });
+        it('should contain only 2 own properties', function() {
```

Comment:
I am wondering if that is required? How would you bypass that in the actual method check? Please provide me with at least one good case, otherwise I think it's meaningless(duplicate).